src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_is_handled_by_slr.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            await Scenario.Define<Context>(c =>
+            {
+                c.Id = Guid.NewGuid();
+                c.TransactionMode = transactionMode;
+            })
+            .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                .When((session, context) => session.SendLocal(new InitiatingMessage
+                {
+                    Id = context.Id
+                }))
+            )
+            .WithEndpoint<ErrorSpy>()
+            .Done(c => c.MessageMovedToErrorQueue)
+            .Repeat(r => r.For<AllDtcTransports>())
+            .Should(c => Assert.IsFalse(c.OutgoingMessageSent, "Outgoing messages should not be sent"))
+            .Run();
+        }
+
+        const string ErrorSpyQueueName = "error_spy_queue";
+
+        class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool MessageMovedToErrorQueue { get; set; }
+            public bool OutgoingMessageSent { get; set; }
+            public TransportTransactionMode TransactionMode { get; set; }
+        }
+
+        class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>((config, context) =>
+                {
+                    var testContext = (Context)context.ScenarioContext;
+
+                    config.UseTransport(context.GetTransportType()).Transactions(testContext.TransactionMode);
+                    config.DisableFeature<FirstLevelRetries>();
+                    config.EnableFeature<SecondLevelRetries>();
+                    config.EnableFeature<TimeoutManager>();
+                    config.Pipeline.Register(new RegisterThrowingBehavior());
+                    config.SendFailedMessagesTo(ErrorSpyQueueName);
+                })
+                .WithConfig<SecondLevelRetriesConfig>(slrConfig =>
+                {
+                    slrConfig.NumberOfRetries = 1;
+                    slrConfig.TimeIncrease = TimeSpan.FromSeconds(1);
+                });
+            }
+
+            class InitiatingHandler : IHandleMessages<InitiatingMessage>
+            {
+                public Context TestContext { get; set; }
+
+                public async Task Handle(InitiatingMessage message, IMessageHandlerContext context)
+                {
+                    if (message.Id == TestContext.Id)
+                    {
+                        await context.Send(ErrorSpyQueueName, new OutgoingMessage { Id = message.Id });
+                    }
+                }
+            }
+        }
+
+        class ErrorSpy : EndpointConfigurationBuilder
+        {
+            public ErrorSpy()
+            {
+                EndpointSetup<DefaultServer>(config => config.LimitMessageProcessingConcurrencyTo(1))
+                    .CustomEndpointName(ErrorSpyQueueName);
+            }
+
+            class ErrorHandler : IHandleMessages<InitiatingMessage>, IHandleMessages<OutgoingMessage>
```

Comment:
The usage of `OutgoingMessage` confused me because we have a type in NSB with that name. Could be come up with another name?